hydra-node/test/Hydra/BehaviorSpec.hs

```diff
@@ -645,6 +615,17 @@ simulatedChainAndNetwork initialChainState = do
           }
     recordAndYieldEvent nodes history chainEvent
 
+  simulateCommit nodes history chainStateVar (party, committed) = do
+    chainEvent <- atomically $ do
+      modifyTVar' chainStateVar advanceSlot
+      cs' <- readTVar chainStateVar
+      pure $
+        Observation
+          { observedTx = OnCommitTx{party, committed}
+          , newChainState = cs'
+          }
+    recordAndYieldEvent nodes history chainEvent
```

Comment:
Should: DRY with `postTx` above.

The act of simulating observation of a Hydra protocol transaction (either `OnCommitTx` or the result of `toOnChainTx`) is identical and can be re-used between `postTx` and `simulateCommit`